apps/service_gather/lib/gather/event/handler.ex

```diff
@@ -13,6 +16,26 @@ defmodule Gather.Event.Handler do
   end
 
   def handle_event(%Brook.Event{type: extract_end(), data: %Extract{} = extract}) do
-    Extraction.Store.delete(extract.dataset_id, extract.subset_id)
+    Extraction.Store.mark_done(extract)
+  end
+
+  def handle_event(%Brook.Event{type: dataset_delete(), data: %Delete{} = delete}) do
+    Logger.debug(fn ->
+      "#{__MODULE__}: Received event #{dataset_delete()}: #{inspect(delete)}"
+    end)
+
+    case Extraction.Store.get!(delete.dataset_id, delete.subset_id) do
+      nil ->
+        Logger.debug("No state to delete")
+        nil
+
+      extract ->
+        if Elsa.topic?(endpoints(), extract.destination),
+          do: Elsa.delete_topic(endpoints(), extract.destination)
+
+        Logger.debug("Deleted Extract destination topic")
```

Comment:
similar comments re: log messages wrapped in anonymous functions and just arbitrarily deleting the topic.